### PR TITLE
Forwarder

### PR DESCRIPTION
**Pull request by janekolszak**
Described in #40

### CODE REVIEW

File: src/utils/model/sql_migrations/000460_interactions_notification.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	is_forwarder_listening boolean;
+   	is_too_big boolean;
+	payload jsonb;
+BEGIN
+    -- Notify only upon L2 changes
+    IF NEW.source != 'redstone-sequencer' THEN
+        RETURN NEW;
+    END IF;
+
+	-- Skip if there's a risk pg_notify would fail
```
Comment (szynwelski) [inline src/utils/model/sql_migrations/000460_interactions_notification.sql:18]: I would correct the indentation for readability

File: src/utils/model/sql_migrations/000460_interactions_notification.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	SELECT pg_notification_queue_usage() > 0.9 INTO is_queue_full;
+	IF is_queue_full THEN
+		-- pg_notify would fail upon full queue, so let's avoid this situation
+		RETURN NEW;
+	END IF;
+
+    -- Neglect big interactions
+    SELECT pg_column_size(NEW.interaction) > 7950 INTO is_too_big;
```
Comment (szynwelski) [inline src/utils/model/sql_migrations/000460_interactions_notification.sql:26]: move it down before the if statement

File: src/utils/model/sql_migrations/000490_sync_state_syncer_notification.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	is_forwarder_listening boolean;
+	payload jsonb;
+BEGIN
+    -- Notify only upon Sequencer's changes
+    IF NEW.name != 'Syncer' THEN
+        RETURN NEW;
+    END IF;
+
+	-- Skip if there's a risk pg_notify would fail
+	SELECT pg_notification_queue_usage() > 0.999 INTO is_queue_full;
```
Comment (szynwelski) [inline src/utils/model/sql_migrations/000490_sync_state_syncer_notification.sql:18]: correct indentation

**Review by szynwelski — CHANGES_REQUESTED** [review]

File: src/forward/arweave-fetcher.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+	if err != nil {
+		return
+	}
+
+	// Merge new LSK into the existing map
+	out = tool.AppendMap(lastSortKeys, newLastSortKeys)
+
+	// Fill in last sort key for each interaction
+	for _, interaction := range interactions {
+		interaction.LastSortKey = lastSortKeys[interaction.ContractId]
```
[the remark this answers is not in the document]
Comment (janekolszak) [reply src/forward/arweave-fetcher.go:174]: Yes, I forgot to update the id in the map. Good catch!

**Pull request merged**